docs/src/developing/on-chain-programs/debugging.md

```diff
@@ -108,3 +108,149 @@ To turn on SBF interpreter trace messages in a local cluster configure the
 `solana_rbpf` level in `RUST_LOG` to `trace`. For example:
 
 `export RUST_LOG=solana_rbpf=trace`
+
+
+## Source level debugging
+
+Source level debugging of on-chain programs written in Rust or C can
```

Comment:
More information about what and isn't available in this environment would be useful.

For example on-chain account data isn't available, instruction data isn't provided, and the syscalls won't work. Right?

So at this point this technique is useful for stepping through program logic that doesn't rely on inputs from the chain

Reply:
True. I'll add a phrase or two on limitations.

Reply:
Sorry, I don't see where this information was added. I'm probably staring right at it and not seeing it!

Reply:
lines 124-130...

Reply:
oh!

> Any information that exists on a blockchain is not
available to the program when executed in rbpf-cli. For example, CPI
calls don't work, reading any information from blockchain doesn't
work, etc.

ok fine, a little hidden for my taste but it's there!